Optional face detection

SumwunDucainest opened the pull request.

SumwunDucainest committed:
Make face detection optional: Add settings to camera proxy

This commit adds a handle to a settings object on Camera Proxy, in order for it
to be able to look up the value of the `faceDetection` setting, which will be used
to enable/disable the face detection feature at runtim.

Signed-off-by: Sumwun Ducainest <sumwun.ducainest@outlook.com>

SumwunDucainest committed:
Make face detection optional: Create the settings

This commit adds the Settings object that will be shared between the QML
frontend and the C++ backend of the app, and give its handle to the CameraProxy.

Signed-off-by: Sumwun Ducainest <sumwun.ducainest@outlook.com>

SumwunDucainest committed:
Make face detection optional: Add a button

This commit adds a face detection toggle in the general panel of the settings
for the user to be able to turn face detection on and off at runtime.

Note that because the settings so far were on the QML side only, the setting for
this button currently lives in a different object.
Ultimately, we might want to merge both setting scopes into one singl shared one.

Signed-off-by: Sumwun Ducainest <sumwun.ducainest@outlook.com>

piggz commented on src/cameraproxy.h:
```diff
@@ -104,6 +105,7 @@ public Q_SLOTS:
 private:
     std::shared_ptr<libcamera::CameraManager> m_cameraManager;
     std::shared_ptr<libcamera::Camera> m_currentCamera;
+    Settings *m_settings;
```
Maybe here should default to nullptr, and check not null when used.  Would be ore robust.

SumwunDucainest committed:
Optional Face detect: Check m_settings

This commit adds a check of the m_settings pointer at runtime in the Camera Proxy.

Signed-off-by: Sumwun Ducainest <sumwun.ducainest@outlook.com>
